modules/rest-util/src/blaze/middleware/fhir/resource.clj

```diff
@@ -95,6 +99,25 @@
       (assoc request :body resource))
     (ba/incorrect "Missing HTTP body.")))
 
+(def ^:private ^Base64$Encoder b64-encoder
+  (.withoutPadding (Base64/getUrlEncoder)))
+
+(defn- get-binary-data [body]
+  (with-open [_ (prom/timer parse-duration-seconds "binary")
+              ^InputStream input body]
+    (.encodeToString b64-encoder (.readAllBytes input))))
```

Comment:
I would put this including the encoder in a function with all the type hints. especially the binding of body to input in with-open is not necessary. it would be possible to just tag body with InputStream. but I usually try to encapsulate all type hints in a simple function